### PR TITLE
Add GitHub workflow for approve/automerge PrettyBot PRs

### PR DESCRIPTION
**Pull request by rosslovas**
# Background

PrettyBot is an automatic process for running a code formatting tool across entire codebases. It's kicked off on a schedule via a TeamCity build.

#project-miso-change-control has introduced a `PRGB` requirement for all changes to our software.

Although we are upping our change control policies, we still want PRs created by bots to auto-merge on a green build.

This PR uses the guidance [here](https://octopushq.atlassian.net/wiki/spaces/RND/pages/2418705819/How+to+Auto+Merge+dependabot+other+bot+related+Pull+Requests+to+my+repositories+protected+branch) to introduce a GitHub Action workflow that auto approves PrettyBot PRs when they are created.

This will ensure PRs raised by the bot will auto-merge on a green build.

# How to review this PR

Quality :heavy_check_mark:

### CODE REVIEW

**Review by paulegradie — APPROVED** [review]
lgtm 👍

File: .github/workflows/prettybot-pull-request-automation.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  # This Allows the GITHUB_TOKEN to auto merge pull requests
+  contents: write
+
+env:
+  PR_URL: ${{github.event.pull_request.html_url}}
+  # By default, GitHub Actions workflows triggered by Dependabot get a GITHUB_TOKEN with read-only permissions.
+  GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+jobs:
+  dependabot:
```
Comment (paulegradie) [inline .github/workflows/prettybot-pull-request-automation.yml:21]:
```suggestion
  prettybot:
```

File: .github/workflows/prettybot-pull-request-automation.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# Increase the access for the GITHUB_TOKEN
+permissions:
+  # This Allows the GITHUB_TOKEN to approve pull requests
+  pull-requests: write
+  # This Allows the GITHUB_TOKEN to auto merge pull requests
+  contents: write
+
+env:
+  PR_URL: ${{github.event.pull_request.html_url}}
+  # By default, GitHub Actions workflows triggered by Dependabot get a GITHUB_TOKEN with read-only permissions.
```
Comment (paulegradie) [inline .github/workflows/prettybot-pull-request-automation.yml:17]:
```suggestion
  # By default, GitHub Actions workflows triggered by PrettyBot get a GITHUB_TOKEN with read-only permissions.
```

**Comment by rosslovas** [comment]
Not rolling this out to public repos for the time being.

**Pull request closed without merging**